### PR TITLE
Task list and task details modals

### PR DESCRIPTION
**Pull request by katiearriagam**
## Changes 
- Task list in the project task tab
- Task details modal when clicking on task row
- Fixed buttons in create task modal
- Fixed indentation in create task blade file. 

## To-do
- Replace placeholders with real data from the DB (waiting on #185)

## Fixes
- #41

### CODE REVIEW

**Comment by hecerinc** [comment]
image

I think this is **much** too space and too little info for the tasks to be full width. Also this makes for a bad experience because you have to scroll through all of your - potentially many - open tasks to get to your closed ones. Should we try a 2-column layout instead?

File: resources/views/projects/details.blade.php [hunk 1]
```diff
@@ -175,13 +196,34 @@ function showTaskModal() {
...
+		onSuccess: function() {
+			console.log($("#new-task-form").serialize());
+			alert('success');
+			event.preventDefault();
+			$.ajax({
+				type: "post",
+				url: "/tasks",
+				data: $("#new-task-form").serialize(),
+				dataType: 'json',
+				success: function (data) {
```
Comment (hecerinc) [inline resources/views/projects/details.blade.php]: Should you be adding the new task to the list?

File: resources/views/projects/details.blade.php [hunk 2]
```diff
@@ -175,13 +196,34 @@ function showTaskModal() {
...
+					console.log(data);
+					alert('success form');
+				},
+				error: function (data) {
+					console.log(data);
+					alert('failed form');
```
Comment (hecerinc) [inline resources/views/projects/details.blade.php]: Maybe we should send a more informative message

File: resources/views/projects/tasks/create.blade.php [hunk 3]
```diff
@@ -1,57 +1,53 @@
 <div class="padded content">
```
Comment (hecerinc) [inline resources/views/projects/tasks/create.blade.php]: Huge humongous padding

image

File: resources/views/projects/tasks/details.blade.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
+<div class="padded content">
```
Comment (hecerinc) [inline resources/views/projects/tasks/details.blade.php]: Huge humonguous padding

image

**Comment by hecerinc** [comment]
Also, you might want to add `cursor: pointer` to the task list items (it's not immediately obvious that they are clickable)

File: app/Project.php [hunk 5]
```diff
@@ -72,13 +72,13 @@ public function collaborators() {
...
 		// TODO: check if supplier_project relationshi exists
-		return $this->collaborators()->where('user_id', $id)->exists();
+		return $this->collaborators()->where('id', $id)->first() != null;
 	}
 
 	// Check if a user is a project stakeholder (as defined in #gh-169)
 	public function isStakeholder($id) {
 		// Check if this is a teacher associated with the course
-		$is_teacher = $this->course()->first()->teachers()->where('user_id', $id)->exists();
+		$is_teacher = $this->course()->first()->teachers()->where('user_id', $id)->first() != null;
```
Comment (hecerinc) [inline app/Project.php]: I'm pretty sure this worked.  I don't think this is necessary

File: resources/views/projects/details.blade.php [hunk 6]
```diff
@@ -284,15 +316,20 @@
...
+	@if($project->isStakeholder(Auth::id()))
+		$(".modal").modal({ transition: "fade up" });
 
+		/* Task details modal */
+		$(".task-title").click(showTaskDetails);
 
-	$(document).ready(function() {
-		$(".ui.fluid.search.dropdown").dropdown();
-		$('#progress').progress();
+		function showTaskDetails() {
```
Comment (hecerinc) [inline resources/views/projects/details.blade.php]: Si no estás usando esto anywhere else, you should make this an anonymous function y pasarselo al click handler

File: resources/views/projects/details.blade.php [hunk 7]
```diff
@@ -306,34 +343,59 @@
...
+			},
+			onSuccess: function(event) {
+				console.log($("#new-task-form").serialize());
+				event.preventDefault();
+				let dueDate = $("#new-task-form #dueDate").val();
+				let isoDueDate = new Date(dueDate).toISOString();
+
+				$.ajax({
+					type: "POST",
+					url: "/tasks",
```
Comment (hecerinc) [inline resources/views/projects/details.blade.php]: Pretty sure this is missing the CSRF token

File: resources/views/projects/details.blade.php [hunk 8]
```diff
@@ -306,34 +343,59 @@
...
 
+		/* Semantic UI form validation */
+		$("#new-task-form").form({
+			fields: {
+				title: ["empty", "maxLength[30]"],
+				description: ["empty", "maxLength[2000]"],
+				dueDate: ["empty"]
+			},
+			onSuccess: function(event) {
+				console.log($("#new-task-form").serialize());
```
Comment (hecerinc) [inline resources/views/projects/details.blade.php]: Let's remove this?

File: resources/views/projects/tasks/index.blade.php [hunk 9]
```diff
@@ -0,0 +1,68 @@
+<div class="ui stackable grid">
```
Comment (hecerinc) [inline resources/views/projects/tasks/index.blade.php]: Por que estos tienen hardcoded data?

File: resources/views/projects/details.blade.php (hunk 1 above)
Comment (katiearriagam) [reply resources/views/projects/details.blade.php]: Sip. coming up on a new pr.

File: resources/views/projects/details.blade.php [hunk 10]
```diff
@@ -306,34 +341,61 @@
 				}
 			}
 		});
-	});
-
 
-	/* Tasks
```
Comment (hecerinc) [inline resources/views/projects/details.blade.php]: Por que borraste mis comentarios? Aprovecho este para decir que esto hay que eventualmente sacarlo a partials por cada tab que hay. Este archivo est&aacute; muy desorganizado, el comentario era un minor attempt de hacerlo m&aacute;s readable

File: resources/views/projects/tasks/create.blade.php [hunk 11]
```diff
@@ -1,56 +1,51 @@
...
+						<label for="description">Description</label>
+						<textarea id="description" name="description" placeholder="e.g. Design the layout for the application in all devices...">{{ old('description') }}</textarea>
+					</div>
 				</div>
-		</div>
+				<div class="sixteen wide column">
+					<!-- Error message -->
+					<div class="ui error message">
+						<h2 class="header">Whoops! Something went wrong.</h2>
+						@if($errors->any())
```
Comment (hecerinc) [inline resources/views/projects/tasks/create.blade.php]: Si est&aacute;s haciendo esto via AJAX exclusivamente entonces esto nunca va a estar poblado (you can safely remove this section)

**Comment by hecerinc** [comment]
Puh-leaaaaaase break it up next time

File: resources/views/projects/details.blade.php (hunk 10 above)
Comment (katiearriagam) [reply resources/views/projects/details.blade.php]: ya los agregué de nuevo, lo que pasa es que además separé por los tipos de usuario pero es cierto que hay que limpiar este archivo

**Pull request merged**